packages/gafl-webapp-service/src/handlers/__tests__/cancel-rp-authentication-handler.spec.js

```diff
@@ -1,18 +1,197 @@
-import cancelRpAuthenticationHandler from '../cancel-rp-authentication-handler'
-import { CONTROLLER } from '../../uri.js'
+import handler from '../cancel-rp-authentication-handler'
+import { CANCEL_RP_IDENTIFY, CANCEL_RP_DETAILS } from '../../uri.js'
+import { addLanguageCodeToUri } from '../../processors/uri-helper.js'
+import { salesApi } from '@defra-fish/connectors-lib'
 
+jest.mock('../../processors/uri-helper.js')
+jest.mock('@defra-fish/connectors-lib')
+jest.mock('../../processors/renewals-write-cache.js', () => ({
+  setUpCacheFromAuthenticationResult: jest.fn().mockResolvedValue(undefined),
+  setUpPayloads: jest.fn().mockResolvedValue(undefined)
+}))
+jest.mock('@defra-fish/business-rules-lib', () => ({
+  validation: {
+    contact: {
+      createBirthDateValidator: () => ({ validateAsync: async () => '1970-01-01' }),
+      createOverseasPostcodeValidator: () => ({ validateAsync: async () => 'AA1 1AA' })
+    }
+  }
+}))
 jest.mock('../../uri.js', () => ({
-  CONTROLLER: { uri: Symbol('controller-uri') }
+  CANCEL_RP_IDENTIFY: { page: 'cancel-rp-identify page', uri: Symbol('cancel-rp-identify-uri') },
+  CANCEL_RP_DETAILS: { uri: Symbol('cancel-rp-details-uri') }
 }))
 
-const getSampleResponseToolkit = () => ({
-  redirectWithLanguageCode: jest.fn()
+const getRequest = (payloadOverride = {}) => {
+  const page = {
+    getCurrentPermission: jest.fn().mockResolvedValue({
+      payload: {
+        referenceNumber: 'ABC123',
+        'date-of-birth-day': '01',
+        'date-of-birth-month': '01',
+        'date-of-birth-year': '1970',
+        postcode: 'AA1 1AA',
+        ...payloadOverride
+      }
+    }),
+    setCurrentPermission: jest.fn()
+  }
+  const status = {
+    getCurrentPermission: jest.fn().mockResolvedValue({}),
+    setCurrentPermission: jest.fn()
+  }
+  const helpers = { page, status }
+  const cache = () => ({ helpers })
+  return { cache }
+}
+
+const getH = () => ({
```

Comment:
Not a well-named variable - h is a response toolkit, so getSampleReponseToolkit would be more informative